Lib/test/test_future.py

```diff
@@ -75,6 +77,20 @@ def test_badfuture10(self):
             from test import badsyntax_future10
         self.check_syntax_error(cm.exception, "badsyntax_future10", 3)
 
+    def test_ensure_flags_dont_clash(self):
+        # bpo-39562: test that future flags and compiler flags doesn't clash
+
+        # obtain future flags (CO_***) from the __future__ module
+        flags = [
+            getattr(__future__, future).compiler_flag
+            for future in __future__.all_feature_names
+        ]
+        # obtain some of the exported compiler flags (PyCF_***) from the ast module
+        flags.extend(
+            getattr(ast, flag) for flag in dir(ast) if flag.startswith("PyCF_")
+        )
```

Comment:
Maybe build a dictionary mapping constant names to values, and then compare:

```
         self.assertCountEqual(set(flags.values()), flags.values()) 
```

For `__future__.all_feature_names`, the name can be retrieved by: f"CO_FUTURE_{name.upper()}".

Maybe it's overkill. But it helped me to understand to write this comment :-D